test/e2e-common.sh

```diff
@@ -210,6 +211,78 @@ function enable_monitoring(){
       --role roles/cloudtrace.agent
 }
 
+function test_authentication_check_for_brokercell() {
```

Comment:
Do we care about leaking anything from this test? If so, we probably need something like a trap to delete the warm up Broker and BrokerCell. We may not care because we expect the cluster to be deleted after this, but just want to make sure we thought about leaking resources.

As-is, if the test fails, then it will leak the warm up Broker.

Reply:
If the test fails, the leaked warm up Broker will be handled when deleting cluster. 

Besides the Broker itself (as the k8s resource), we also need to take care about the PubSub subscription and PubSub topic (as the Cloud resource) which were created by the controller when reconciling the Broker. If the test fails, the deletion of remaining pubsub subscription and pubsub topic would rely on the delete_topics_and_subscriptions function (https://github.com/google/knative-gcp/blob/ce080cf13a8d42e0588684f140c08d0f94666f85/test/e2e-common.sh#L180-L195). That is, every time when start to run the test, clean up the subscription and topic from last run.

That's the two kind of the resource which might be leak, and the way to clean up them, AFAIK.

Reply:
Thank you for explaining it!.